mesonbuild/coredata.py

```diff
@@ -1025,22 +1026,7 @@ def major_versions_differ(v1: str, v2: str) -> bool:
 
 def load(build_dir: str) -> CoreData:
     filename = os.path.join(build_dir, 'meson-private', 'coredata.dat')
-    load_fail_msg = f'Coredata file {filename!r} is corrupted. Try with a fresh build tree.'
-    try:
-        with open(filename, 'rb') as f:
-            obj = pickle.load(f)
-    except (pickle.UnpicklingError, EOFError):
-        raise MesonException(load_fail_msg)
-    except (ModuleNotFoundError, AttributeError):
-        raise MesonException(
-            f"Coredata file {filename!r} references functions or classes that don't "
-            "exist. This probably means that it was generated with an old "
-            "version of meson.")
-    if not isinstance(obj, CoreData):
-        raise MesonException(load_fail_msg)
-    if major_versions_differ(obj.version, version):
-        raise MesonVersionMismatchException(obj.version, version)
-    return obj
+    pickle_load(filename, 'Coredata', CoreData)
```

Comment:
This needs to return the loaded coredata...

Reply:
argh, of course, how did I miss that! Fixed.